app/uk/gov/hmrc/gform/sharedmodel/form/FormData.scala

```diff
@@ -30,6 +31,8 @@ case class FormData(fields: List[FormField]) {
 
   def ++(other: FormData): FormData =
     FormData((toData ++ other.toData).map { case (k, v) => FormField(k, v) }.toList)
+
+  def toHash: Option[String] = Some(CryptoUtil.sha256Hash(Json.toJson(FormData(fields)).toString()))
```

Comment:
I rather not call `.toString`. There `Json.stringify` which seems like a better implementation.

Reply:
I don't see any reason for this to return an `Option`

Reply:
Also what about renaming `toHash` to `fingerprint` ?